InfiniLink/Music/MusicController.swift

```diff
@@ -63,11 +72,43 @@ class MusicController {
 		return currentSongInfo
 	}
 	
-	func updateMusicInformation(songInfo: MusicController.songInfo) {
-		let bleWriteManager = BLEWriteManager()
+    func updateMusicInformation(songInfo: MusicController.songInfo) {
+        let bleWriteManager = BLEWriteManager()
 		let songInfo = getCurrentSongInfo()
 		
 		bleWriteManager.writeToMusicApp(message: songInfo.trackName, characteristic: bleManager.musicChars.track)
 		bleWriteManager.writeToMusicApp(message: songInfo.artistName, characteristic: bleManager.musicChars.artist)
+        
+        var playbackTime = musicPlayer.currentPlaybackTime; if playbackTime == musicPlayer.nowPlayingItem!.playbackDuration {playbackTime = 0.0}
```

Comment:
I forgot that I had deleted Apple Music from my phone when I first ran your PR, and experienced a crash on this line, so there should be a check on this force unwrap. I think probably the best place for it is at the updateMusicInformation call in controlMusic above, as we don't really need to call this function if there's nothing to call it against: 

```
if musicPlayer.nowPlaying != nil {
	updateMusicInformation(songInfo: getCurrentSongInfo())
}
```

Reply:
Should be fixed now.